docs/jwst/straylight/main.rst

```diff
@@ -19,20 +15,57 @@ in the MRS slice
 
-The source of the MIRI MRS stray-light has been identified as being caused
+The chief source of the MIRI MRS stray-light appears to be  caused
 by scattering in optical components within the SMO. The stray-light is
 manifested as a signal that extends in the detector row direction. Its
 magnitude is proportional to that of bright illuminated regions of the
 spectral image, at a ratio that falls with increasing wavelength,
-from about 2 % in Channel 1A to undetectably low levels longward of Channel 2B.
+from about 1 % in Channel 1A to undetectable low levels long-ward of Channel 2B.
+Two components of the stray-light have been observed, a smooth and a structured
+distribution. 
 
 Algorithm
 ---------
+The basic idea of the stray-light removal algorithm is to deal with the 
+smooth component of the stray-light only. Due to the extended nature of the
+stray-light we use the detected signal in the slice gaps, where nominally no photons
+should it the detectors, and assume that all detected light is the stray-light. 
+Using this measurement, we can interpolate the gap flux within the slice to
+estimate the amount of the stray-light in the slice. 
 
-The MIRI MRS algorithm uses a stray-light MASK reference file to determine
-which pixels are science pixels and which pixels fall in-between the
-slices. Each illuminated pixel on the array is has a signal that is the
-sum of direct illumination and the scattering from neighboring areas.
-Only the pixels located between the slices are areas of indirect illumination.
-The  illumination on the inter-slice pixels are used  to determine a
-stray-light component to subtract from each science pixel.
+There are two possible algorithms in the stray-light step. The first algorithm is
+a more simplistic approach by dealing with the stray-light estimation row-by-row
+and interpolating the gap flux linearly. An intermediate stray-light map is 
+generated row-by-row and then this map is further smoothed to remove row-by-row
+variations. This algorithm uses a stray-light mask reference file that contains
+1s for gap pixels and 0s for science pixels.  
+
+Given the extended nature of the smooth component of the MRS stray-light, it
+is obvious that a row-by-row handling of the stray-light could be replaced
+by a two-dimensional approach such that not additional smoothing is required.
+For the second algorithm we improve the technique by using the Modified Shepard's
+Method to interpolate the gap fluxes two dimensionally. 
+
+The second algorithm uses the MRS distortion reference file. The first extension
+of this file contains the "Slice_Number" and consists of a detector pixel map 
+indicating the slice numbers or in case of a non-science pixel, the value zero. 
+Consequently, the gap pixels can be easily found by masking all pixels that are 
+non-science.The assign_wcs step takes the "Slice_Number" information and stores it in
+a regions file.  This second algorithm takes each slice pixel and determines the 
```

Comment:
where is the regions file? Is it really a file (on disk) that's generated by assign_wcs or is it just an additional data structure of some kind added to the data model?